lib/src/types.ts

```diff
@@ -431,6 +460,28 @@ export function possibleRootTypes(
       []
     );
   }
+
+  if (isIntersectionType(type)) {
+    const concreteTypes = type.types.reduce<ConcreteType[]>((acc, curr) => {
+      return acc.concat(possibleRootTypes(curr, typeTable));
+    }, []);
+    // if they are all objects then reduce them into one object
+    // to allow for proper serialization
+    if (concreteTypes.every(isObjectType)) {
+      return [
+        {
+          kind: TypeKind.OBJECT,
+          properties: concreteTypes.reduce<ObjectPropertiesType[]>(
+            (acc, curr) => {
+              return acc.concat(...curr.properties);
+            },
+            []
+          )
+        }
+      ];
+    }
+    return concreteTypes;
```

Comment:
Let's assume the type parser should ensure concrete types of intersections are object types and throw an error here otherwise.